src/FlipMove.js

```diff
@@ -238,7 +241,7 @@ class FlipMove extends Component {
     // we need to reset the transition, so that the NEW shuffle starts from
     // the right place.
     this.state.children.forEach((child) => {
-      const { domNode } = this.childrenData[child.key];
+      const { domNode } = this.childrenData[child.key] || {};
```

Comment:
Yes, there was another issue with Preact, I think it resolves ref callbacks a bit later than React does, but this seemed to fix the issue.